### ISSUE TITLE
Update styles in Android test app

### ISSUE THREAD

**Issue by ljbade** [mapbox/mapbox-gl-native#1264]
Time to update Android to use the same styles as iOS:
* Light
* Dark
* Emerald
* Mapbox Streets
* ~~Outdoors~~

A few questions @bleege and @incanus:
Do I remove the other styles? (e.g. bright/pencil)
What order should the styles be in the drop down (and which one is the default) to match iOS?
Do any of these styles have switchable classes like the day/night in Outdoors?

**Comment by 1ec5** [comment]
> Do any of these styles have switchable classes like the day/night in Outdoors?

We’ve formalized a styleClasses API on iOS, but if any of the bundled styles have toggleable classes, their designers have kept it a secret. :smiley: Note that the Satellite style does have classes for labes/roads and contours, which are akin to the “hybrid” style offered by Google and Apple. We aren’t shipping with Satellite just yet but will once #963 is unblocked.

**Comment by bleege** [comment]
> Do I remove the other styles? (e.g. bright/pencil)

Yes please.

> What order should the styles be in the drop down (and which one is the default) to match iOS?

iOS is using Emerald as it's default style.  Streets will probably be the second one.  After that the order shouldn't matter too much.

**Comment by incanus** [comment]
Clarification here: we are only pushing on four styles — not Outdoors. IIRC @samanpwbb said that it would need some special attention and isn't part of the current push. For context, Outdoors was our June '14 launch style. 

> Do I remove the other styles? (e.g. bright/pencil)

Here are the styles in the iOS demo app: 

```objc
static NSArray *const kStyleNames = @[
    @"Mapbox Streets",
    @"Emerald",
    @"Light",
    @"Dark",
    @"Bright",
    @"Basic",
    @"Outdoors",
    @"Satellite",
];
```

[`mapbox-gl-native/ios/app/MBXViewController.mm#L13-L22`](https://github.com/mapbox/mapbox-gl-native/blob/9f13392de0a89fb727f35a1842f775c39da28249/ios/app/MBXViewController.mm#L13-L22)

And here are the styles shown in Sirius: 

```swift
let SiriusShownStyles = [ "emerald-v7", "light-v7", "dark-v7", "mapbox-streets-v7" ]
```

[`sirius-ios/Sirius/Constants.swift#L8`](https://github.com/mapbox/sirius-ios/blob/ed1edfdb1823a32b383a3a8aa6c6cbf63760be47/Sirius/Constants.swift#L8)

In short, show all of the styles that don't break GL in the demo app (e.g. get rid of Pencil) and just show the four we're sprinting on / will launch with in Sirius. 

> Do any of these styles have switchable classes like the day/night in Outdoors?

The new ones don't to my knowledge, but Outdoors for example does. Here's how to tell: 

```css
    "paint": {
      "background-color": "@land"
    },
    "paint.night": {
      "background-color": "@land_night"
    }
```

[`mapbox-gl-styles/styles/outdoors-v7.json#L196-L201`](https://github.com/mapbox/mapbox-gl-styles/blob/3b169f4e057393fa0d4a9831a1061e7a0054117c/styles/outdoors-v7.json#L196-L201)

You will always see `paint` in a style, but the presence of `paint.<class>` indicates class toggles.

**Comment by incanus** [comment]
Also, if not otherwise clear, both the GL app and Sirius now default to Mapbox Streets instead of Emerald.

**Comment by ljbade** [comment]
Perfect, thanks for the details @incanus working on this today.

**Comment by ljbade** [comment]
I am picking this up again and working on updating the `android-styles` branch to latest master.

**Comment by ljbade** [comment]
@bleege @incanus @BergWerkGIS @mb12 Can you guys give https://github.com/mapbox/mapbox-gl-native/compare/android-styles a test to see if it fixes the style loading issues in #1468 ?

@bleege @incanus Can you also let me know if you are happy with the new list of styles:
- Mapbox Streets (default)
- Emerald
- Light
- Dark

**Comment by incanus** [comment]
List looks great; will test branch.

**Comment by incanus** [comment]
Ok, this definitely gets me past the style issues, but ends with the same crash. 

```
05-15 17:46:21.229     995-1012/com.mapbox.mapboxgl.testapp A/libc﹕ Fatal signal 11 (SIGSEGV), code 1, fault addr 0x0 in tid 1012 (apboxgl.testapp)
```
